src/main/java/com/github/cameltooling/lsp/internal/parser/CamelYamlDSLParser.java

```diff
@@ -220,4 +254,46 @@ private String extractUriFromYamlData(String line) {
 		}
 		return null;
 	}
+
+	private int extractLines(TextDocumentItem textDocumentItem, Position position, String whenToStop,
+							 StringBuilder lines, int pos, boolean cleanNewLine) {
+		return extractLines(textDocumentItem, position, new String[]{whenToStop}, lines, pos, cleanNewLine);
+	}
+
+	private int extractLines(TextDocumentItem textDocumentItem, Position position, String[] whenToStopArray,
+							 StringBuilder lines, int pos, boolean cleanNewLine) {
+		// If it is a multi-line, cleanNewLine decides if we convert it to one very long single line
+		for (Integer lineNo = position.getLine(); lineNo >=0; lineNo--) {
+			String tempLine = parserFileHelperUtil.getLine(textDocumentItem, lineNo);
+
+			for (String whenToStop : whenToStopArray) {
+				if (tempLine.stripLeading().startsWith(whenToStop)) {
+					// Remove the potential '>' character of multiline
+					if (tempLine.stripLeading().substring(whenToStop.length() + 1).stripLeading().startsWith(">")) {
+						tempLine = tempLine.substring(0, tempLine.indexOf(">"));
+					}
+					lineNo = -1; // Stop after this iteration, do not continue the loop
+					break;
+				}
+			}
+
+			// Prepend the content
+			if (cleanNewLine) {
+				// If it is multiline, we have to move the position forward
+				if (lines.length() > 0) {
+					final var tempLineStripped = tempLine.stripLeading();
+					pos += tempLineStripped.length();
+					lines.insert(0, tempLineStripped);
```

Comment:
We strip the line from all the previous whitespace because we are unifying all the string into one big string. And the trailing spaces won't make sense on this one big string.